### ISSUE TITLE
Electron TLS module on Windows does not send data in a timely manner

### ISSUE THREAD

**Issue by ItsOnlyBinary** [electron/electron#27633]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v11.2.2, 12.0.0-beta.22, 13.0.0-nightly.20210203
* **Operating System:**
  * Windows 10 (20H2)
* **Last Known Working Electron version:**
  * Unknown

### Expected Behavior
when using write() on a `tls` socket, I expect the data to be sent in a timely manor and the callback executed. The same as a `net` socket behaves or `tls` socket on mac os

```
server listening 127.0.0.1:5734
socket open
data: test 1\r\n [8 bytes]
data: test 2\r\n [8 bytes]
data: test 3\r\n [8 bytes]
data: test 4\r\n [8 bytes]
data: test 5\r\n [8 bytes]
data: test 6\r\n [8 bytes]
data: test 7\r\n [8 bytes]
data: test 8\r\n [8 bytes]
sending test
data: test 9\r\n [8 bytes]
data: test 10\r\n [9 bytes]
data: test 11\r\n [9 bytes]
```

### Actual Behavior
on the first write() the data is sent to the remote but the callback is not called, subsequent data appears to get buffered until data is received from the remote

```
server listening 127.0.0.1:5734
socket open
data: test 1\r\n [8 bytes]
sending test
data: test 2\r\ntest 3\r\ntest 4\r\ntest 5\r\ntest 6\r\ntest 7\r\ntest 8\r\ntest 9\r\ntest 10\r\n [73 bytes]
```

### To Reproduce

To assist in reproducing this I've made a nodejs tls server and electron app with tls client, this can be found here:

https://github.com/ItsOnlyBinary/test-electron

Instructions are also within the repo's README.md

####  Cloning the test server and app
```sh
$ git clone https://github.com/ItsOnlyBinary/test-electron -b master
$ cd test-electron
```

####  Running the tls server in terminal 1, within `socket-listener` directory
```sh
$ cd socket-listener
$ node index.js
```

####  Running the electron app in terminal 2, within `electron-app` directory
```sh
$ cd electron-app
$ electron .
```

### Additional Information
I've also tried creating the socket within the main process, and using ipcMain/ipcRenderer to communicate with it, unfortunately this produced the same results.

I could not reproduce this issue in only nodejs

**Comment by ckerr** [comment]
@ItsOnlyBinary, thanks for reaching out and reporting this bug.


Might be windows-only. Testing on Ubuntu 20.10 gives me similar output to the macOS output listed in the description under "Expected Behavior":

```
server listening 127.0.0.1:5734
socket open
data: test 1\r\n [8 bytes]
data: test 2\r\n [8 bytes]
data: test 3\r\n [8 bytes]
data: test 4\r\n [8 bytes]
data: test 5\r\n [8 bytes]
data: test 6\r\n [8 bytes]
data: test 7\r\n [8 bytes]
data: test 8\r\n [8 bytes]
data: test 9\r\n [8 bytes]
data: test 10\r\n [9 bytes]
data: test 11\r\n [9 bytes]
data: test 12\r\n [9 bytes]
data: test 13\r\n [9 bytes]
sending test
```

Tested on Ubuntu 20.10 with v12.0.0-beta.20 with https://github.com/ItsOnlyBinary/test-electron as per the instructions in this issue's description, then looking at the console output from the tls server in terminal 1.

**Comment by ItsOnlyBinary** [comment]
This appears to be fixed in electron v21.1.0
